crates/net/network/src/session/active.rs

```diff
@@ -84,6 +84,9 @@ pub(crate) struct ActiveSession {
     pub(crate) internal_request_tx: Fuse<ReceiverStream<PeerRequest>>,
     /// All requests sent to the remote peer we're waiting on a response
     pub(crate) inflight_requests: FnvHashMap<u64, InflightRequest>,
+    /// Counting semaphore notifies [`crate::transactions::TransactionsManager`] of total inflight
+    /// requests.
+    pub(crate) inflight_requests_semaphore_tx: watch::Sender<usize>,
```

Comment:
do we intend to subscribe to value changes?

if not then this could also be an Atomic?

Reply:
not sure we want to keep this and if we do we'll use `Semaphore` type https://github.com/paradigmxyz/reth/pull/6012#discussion_r1447751264